tests/integration/replication.tcl

```diff
@@ -1355,3 +1355,34 @@ start_server {tags {"repl" "external:skip"}} {
         assert_equal "PONG" [r ping]
     }
 }
+
+start_server {tags {"repl external:skip"}} {
+    set master [srv 0 client]
+    set master_host [srv 0 host]
+    set master_port [srv 0 port]
+    $master debug SET-ACTIVE-EXPIRE 0
+    start_server {} {
+        set slave [srv 0 client]
+        $slave debug SET-ACTIVE-EXPIRE 0
+        $slave slaveof $master_host $master_port
+
+        test "Tes replication with lazy expire" {
+            # wait for replication to be in sync
+            wait_for_condition 50 100 {
+                [lindex [$slave role] 0] eq {slave} &&
+                [string match {*master_link_status:up*} [$slave info replication]]
+            } else {
+                fail "Can't turn the instance into a replica"
+            }
+
+            $master sadd s foo
+            $master expire s 1
+            after 1000
+            $master sadd s foo
+            assert_equal 1 [$master wait 1 1]
+
+            assert_equal "set" [$master type s]
+            assert_equal "set" [$slave type s]
```

Comment:
before this commit the replica didn't have `foo` at all?

Reply:
The replica did not have the set at all.